src/Observable.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { Observable } from './Observable'
+
+const impossibru = function (done, msg): () => void {
+    return (): void => {
+        expect(`${msg} WILL NOT HAPPEN`).toBe(''); setTimeout(done)
+    }
+}
+
+describe('Observable', () => {
+    it('can filter messages', done => {
+        const tester = value => {
+            expect(value).toBe(3)
+            setTimeout(done)
+        }
+        Observable.of(1, 2, 3).filter(value =>
+            value > 2
+        ).subscribe(value => {
+            tester(value)
+        }, impossibru(done, 'Error sub trigger'), impossibru(done, 'Completion sub trigger'))
+    })
+
+    it('can reduce messages', done => {
+        const tester = value => {
+            expect(value).toBe(6)
+            setTimeout(done)
+        }
+        Observable.of(1, 2, 3).reduce((acc, value) =>
+            acc + value
+        ).subscribe(value => {
+            tester(value)
+        }, impossibru(done, 'Error sub trigger'), impossibru(done, 'Completion sub trigger'))
+    })
+
+    it('can map messages', () => {
+        let next = null
+        const values = []
+
+        const subscription = new Observable(observer => {
```

Comment:
@arturkulig  Pewnie tak. Chciałem mieć sprawdzanie wartości po każdym wykonaniu, a nie wiem jak to zrealizować z `Observable.of`? Czy tak jak w BaseObservable.spec zrobić coś w stylu firstValueTester / secondValueTester / thirdValueTester? Czy można jakoś prościej?